tools/functional-tester/etcd-tester/lease_stresser.go

```diff
@@ -0,0 +1,373 @@
+}
+
+func (ls *leaseStresser) createLeaseAndKeepAlive(ctx context.Context) (int64, error) {
+	leaseID, err := ls.createLease(ctx)
+	if err != nil {
+		plog.Printf("created lease ERROR %v", err)
+		return leaseID, err
+	}
+
+	go ls.keepLeaseAlive(ctx, leaseID)
+	return leaseID, nil
+}
+
+func (ls *leaseStresser) createLease(ctx context.Context) (int64, error) {
+	resp, err := ls.lc.LeaseGrant(ctx, &pb.LeaseGrantRequest{
+		TTL: TTL,
+	})
+
+	if err != nil {
+		plog.Printf("creating lease ERROR %v", err)
+		return 0, err
+	}
+
+	return resp.ID, nil
+}
+
+func (ls *leaseStresser) keepLeaseAlive(ctx context.Context, leaseID int64) {
+	stream, err := ls.lc.LeaseKeepAlive(ctx, grpc.FailFast(false))
+	if err != nil {
+		plog.Printf("keepLeaseAlive ERROR %v", err)
+		return
+	}
+
+	for {
+		select {
+		case <-time.After(1 * time.Second):
+		case <-stream.Context().Done():
+			plog.Printf("Stresser %v keepLeaseAlive stream closed", ls.id)
+			return
+		}
+		err = stream.Send(&pb.LeaseKeepAliveRequest{ID: leaseID})
+		if err != nil {
+			plog.Printf("keepLeaseAlive stream send lease %v ERROR %v ", leaseID, err)
+			continue
+		}
+
+		respRC, err := stream.Recv()
+
+		if err != nil {
+			plog.Printf("keepLeaseAlive stream recieve lease %v stream ERROR %v ", leaseID, err)
+			continue
+		}
+		// lease expires after TTL become 0
+		// don't send keepalive if the lease has expired
+		if respRC.TTL <= 0 {
+			plog.Printf("keepLeaseAlive: lease %v has expired ", respRC.ID)
+			return
+		}
+	}
+}
+
+// attachKeysWithLease function attaches keys to the lease.
+// the format of key is the concat of leaseID + '_' + '<order of key creation>'
+// e.g 5186835655248304152_0 for first created key and 5186835655248304152_1 for second created key
+func (ls *leaseStresser) attachKeysWithLease(ctx context.Context, leaseID int64) error {
+	for j := 0; j < ls.numOfKeysPerLease; j++ {
+		_, err := ls.kvc.Put(ctx, &pb.PutRequest{
+			Key:   []byte(fmt.Sprintf("%d%s%d", leaseID, "_", j)),
+			Value: []byte(fmt.Sprintf("bar")),
+			Lease: leaseID,
+		}, grpc.FailFast(false))
+
+		if err != nil {
+			plog.Printf("attach keys to lease %v ERROR %v", leaseID, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (ls *leaseStresser) randomlyDropLease(ctx context.Context, leaseID int64) bool {
+	// randomlyDropLease drops the lease only when the rand.Int(2) returns 1.
+	// This creates a 50/50 percents chance of dropping a lease
+	// if rand.Intn(2) != 0 {
+	// 	return false
+	// }
+	// always drop lease for now
+	return false
+	_, err := ls.lc.LeaseRevoke(ctx, &pb.LeaseRevokeRequest{
+		ID: leaseID,
+	}, grpc.FailFast(false))
+
+	if err != nil {
+		plog.Printf("fail to droplease %v ERROR %v", leaseID, err)
+		return false
+	}
+	return true
+}
+
+var (
+	noPrefixEnd = []byte{0}
+)
+
+func (ls *leaseStresser) Cancel() {
+	// dont do anything
+}
+
+func (ls *leaseStresser) Report() (int, int) {
+	return ls.success, ls.failure
+}
+
+func getPrefix(key []byte) []byte {
```

Comment:
Since GetPrefixRangeEnd returns `string` not `[]byte` which is what `RangeRequest` takes, I should just convert the returned string to `[]byte` right?